src/Microsoft.Framework.Runtime.Roslyn/RoslynProjectReference.cs

```diff
@@ -81,28 +73,36 @@ public Assembly Load(IAssemblyLoadContext loadContext)
 
                 if (_supportsPdbGeneration.Value)
                 {
-                    emitResult = CompilationContext.Compilation.Emit(assemblyStream, pdbStream: pdbStream, manifestResources: resources);
+                    emitResult = BeforeCompileContext.Compilation.Emit(assemblyStream, pdbStream: pdbStream, manifestResources: resources);
                 }
                 else
                 {
                     Logger.TraceWarning("PDB generation is not supported on this platform");
-                    emitResult = CompilationContext.Compilation.Emit(assemblyStream, manifestResources: resources);
+                    emitResult = BeforeCompileContext.Compilation.Emit(assemblyStream, manifestResources: resources);
                 }
 
                 sw.Stop();
 
                 Logger.TraceInformation("[{0}]: Emitted {1} in {2}ms", GetType().Name, Name, sw.ElapsedMilliseconds);
 
-                var diagnostics = CompilationContext.Diagnostics.Concat(
+                var diagnostics = BeforeCompileContext.Diagnostics.Concat(
                     emitResult.Diagnostics);
 
-                var afterCompileContext = new AfterCompileContext(CompilationContext, diagnostics)
+                var afterCompileContext = new AfterCompileContext()
                 {
+                    ProjectContext = BeforeCompileContext.ProjectContext,
+                    Compilation = BeforeCompileContext.Compilation,
                     AssemblyStream = assemblyStream,
-                    SymbolStream = pdbStream
+                    SymbolStream = pdbStream,
+                    Diagnostics = new List<Diagnostic>()
                 };
 
-                foreach (var m in CompilationContext.Modules)
+                foreach (var diagnostic in diagnostics)
```

Comment:
Can't you just do `Diagnostics = new List<Diagnostic>(diagnostics)` in the object initializer?